lua/bufferline/state.lua

```diff
@@ -211,5 +218,37 @@ function state.set_offset(width, text, hl)
   require'bufferline.api'.set_offset(width, text, hl)
 end
 
+--- Restore the buffers
+--- @param buffer_data {name: string, pinned: boolean}[]
+function state.restore_buffers(buffer_data)
```

Comment:
I moved `restore_buffers` here, because honestly there wasn't anything having to do with `render` left in the function. It only modifies `state`, and doesn't even `render.update` afterward. I feel like it belongs here (and moving it doesn't change any other code, except the `SessionSavePre` string)